schema.cc

```diff
@@ -656,6 +658,145 @@ std::ostream& operator<<(std::ostream& os, const schema& s) {
     return os;
 }
 
+std::ostream& map_as_cql_param(std::ostream& os, const std::map<sstring, sstring>& map, bool first = true) {
+    for (auto i: map) {
+        if (first) {
+            first = false;
+        } else {
+            os << ",";
+        }
+        os << "'" << i.first << "': '" << i.second << "'";
+    }
+
+    return os;
+}
+
+std::ostream& column_definition_as_cql_key(std::ostream& os, const column_definition & cd) {
+    os << cd.name_as_cql_string();
+    os << " " << cd.type->cql3_type_name();
+
+    if (cd.kind == column_kind::static_column) {
+        os << " STATIC";
+    }
+    return os;
+}
+
+static sstring remove_index_suffix(const sstring& name) {
+    return name.substr(0, name.size() - 6); // remove the _index suffix from an index name;
+}
+
+static bool is_secondary_index(const utils::UUID& id, const schema& s) {
+    return  service::get_local_storage_service().db().local().find_column_family(id).get_index_manager().is_global_index(s);
+}
+std::ostream& schema::describe(std::ostream& os) const {
+    os << "CREATE ";
+    int n = 0;
+
+    if (is_view()) {
+        if (is_secondary_index(view_info()->base_id(), *this)) {
+            os << "INDEX " << remove_index_suffix(cf_name()) << " ON " << ks_name() << "." << view_info()->base_name() << "(";
+            for (auto& pk : partition_key_columns()) {
+                if (n++ != 0) {
+                    os << ", ";
+                }
+                os << pk.name_as_cql_string();
+            }
+            os <<");\n";
+            return os;
+        } else {
+            os << "MATERIALIZED VIEW " << ks_name() << "." << cf_name() << " AS\n";
```

Comment:
it should work, but I'll verify.

Reply:
I don't see how this can work without being fixed. If cf_name is FooBar (capitalized like this), the only correct way to create is with the CQL command:
```
CREATE TABLE "FooBar"
```
Like this - with double quotes. Without the double quotes, it will just create a table called foobar - and this is not the table you wanted to create. Fixing this is trivial - you can use the existing function, `cql3::util::maybe_quote(sstring(cf_name))` to do this CQL quoting if necessary.

Reply:
I already tested it again and validate that there is an issue with the table name. The next version will take care of that and I have tests that validate that it's correct.

While the parameters are quoted correctly, that table/ks wheren't (but it will be ok)

Reply:
Right. The column names you did quote correctly - just not the table names (the index name, view name, and table name, can all have this issue).